### PR TITLE
Add type checking with mypy

### PR DESCRIPTION
**Pull request by araffin**
## Description

Counterpart of https://github.com/DLR-RM/stable-baselines3/pull/1143

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,9 +8,13 @@ pytest:
...
 	python -m pytest -v tests/test_enjoy.py -k trained_agent --color=yes
 
-# Type check
-type:
-	pytype -j auto rl_zoo3/ tests/ scripts/ -d import-error
+pytype:
+	pytype -j auto ${LINT_PATHS} -d import-error
+
+mypy:
+	mypy ${LINT_PATHS} --install-types --non-interactive
```
Comment (qgallouedec) [inline Makefile:15]: Should we also add `--install-types --non-interactive` in sb3 and sb3-contrib?

**Comment by qgallouedec** [comment]
https://github.com/DLR-RM/rl-baselines3-zoo/pull/331/commits/de67905f0085d8c6163a6f74f57d1a4ad8009fd2: https://github.com/python/mypy/issues/2427

**Review by qgallouedec — APPROVED** [review]

**Comment by araffin** [comment]
It's weird, mypy check passes locally, I guess it has something to do with the way we package SB3 / SB3 contrib, maybe the stubs are missing? (locally I have them installed in editable mode)

File: Makefile (hunk 1 above)
Comment (araffin) [reply Makefile:15]: I would only if it was needed.

**Comment by qgallouedec** [comment]
Same here

**Comment by araffin** [comment]
I can reproduce some errorr locally: `pip install .` vs `pip install -e .` (for sb3-contrib) with python 3.8 and latest mypy version.

and surprisingly, I've got more errors on my laptop that on the PC I used to do the PR.

**Comment by qgallouedec** [comment]
> I can reproduce some errorr locally: `pip install .` vs `pip install -e .` (for sb3-contrib) with python 3.8 and latest mypy version.
> 
> and surprisingly, I've got more errors on my laptop that on the PC I used to do the PR.

Have you checked that the sb3/sb3-contrib versions match?

**Comment by araffin** [comment]
> Have you checked that the sb3/sb3-contrib versions match?

yes, I'm using master version of both

**Comment by qgallouedec** [comment]
> Same here

Actually, no, I just forgot to stage setup.cfg in my last commit.

**Review by qgallouedec — APPROVED** [review]

**Pull request merged**